metadata-ingestion/src/datahub/ingestion/source/cassandra/cassandra_api.py

```diff
@@ -0,0 +1,175 @@
+                ssl_context = SSLContext(PROTOCOL_TLSv1_2)
+                ssl_context.verify_mode = CERT_NONE
+                auth_provider = PlainTextAuthProvider(
+                    username=self.config.username, password=self.config.password
+                )
+
+            cluster = Cluster(
+                [self.config.contact_point],
+                port=self.config.port,
+                load_balancing_policy=None,
+            )
+
+            if auth_provider:
+                cluster = Cluster(
+                    [self.config.contact_point],
+                    port=self.config.port,
+                    auth_provider=auth_provider,
+                    ssl_context=ssl_context,
+                    load_balancing_policy=None,
+                )
+            session = cluster.connect()
+            return session
+        except Exception as e:
+            logger.error("Failed to authenticate to Cassandra", exc_info=True)
+            self.report.report_failure(
+                message="Failed to authenticate to Cassandra", exc=e
+            )
+            raise
+
+    def get_keyspaces(self) -> List:
+        """Fetch all keyspaces."""
+        try:
+            keyspaces = self.cassandra_session.execute(
+                CassandraQueries.GET_KEYSPACES_QUERY
+            )
+            keyspaces = sorted(
+                keyspaces,
+                key=lambda k: getattr(
+                    k, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["keyspace_name"]
+                ),
+            )
+            return keyspaces
+        except Exception:
+            logger.error("Failed to fetch keyspaces", exc_info=True)
+            self.report.warning(
+                message="Failed to fetch keyspaces",
+            )
+            return []
+
+    def get_tables(self, keyspace_name: str) -> List:
+        """Fetch all tables for a given keyspace."""
+        try:
+            tables = self.cassandra_session.execute(
+                CassandraQueries.GET_TABLES_QUERY, [keyspace_name]
+            )
+            tables = sorted(
+                tables,
+                key=lambda t: getattr(
+                    t, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["table_name"]
+                ),
+            )
+            return tables
+        except Exception:
+            logger.error(
+                f"Failed to fetch tables for keyspace {keyspace_name}", exc_info=True
+            )
+            self.report.warning(
+                message="Failed to fetch tables for keyspace",
+                context=f"{keyspace_name}",
+            )
+            return []
+
+    def get_columns(self, keyspace_name: str, table_name: str) -> List:
+        """Fetch all columns for a given table."""
+        try:
+            column_infos = self.cassandra_session.execute(
+                CassandraQueries.GET_COLUMNS_QUERY, [keyspace_name, table_name]
+            )
+            column_infos = sorted(column_infos, key=lambda c: c.column_name)
```

Comment:
Does cassandra provide column order in its system view of columns? Some systems do provide it, and if so we should honour it rather than sorting by name.

Reply:
It only applied for PRIMARY KEY not for others.